### ISSUE TITLE
Plugin exports do not match types

### ISSUE THREAD

**Issue by soren121** [import-js/eslint-plugin-import#3212]
The types added in #3097 do not match with what the plugin currently exports.

The type definitions declare a default export, which this plugin does not have. The ESLint documentation [recommends](https://eslint.org/docs/latest/extend/plugins#creating-a-plugin) a default export. 

Not providing this export, and then declaring that it is provided, can cause user confusion. For instance, I would expect the following to work:

```js
import importPlugin from 'eslint-plugin-import';
import { defineConfig } from 'eslint/config';

export default defineConfig(
  {
    plugins: {
      import: importPlugin,
    },
    rules: {
      'import/named': 'error',
    }
  },
  {
    files: ['src/**/*.{ts,tsx}'],
    extends: [importPlugin.flatConfigs.typescript]
  }
);
```

TypeScript compiles this without any warning, but ESLint will produce this error:

```
Oops! Something went wrong! :(

ESLint: 9.36.0

ConfigError: Config "UserConfig[1] > import/typescript": Key "plugins": Cannot redefine plugin "import".
```

**Comment by soren121** [comment]
I have `allowSyntheticDefaultImports` and `esModuleInterop` both set to true in my tsconfig.json.

> The package is CJS, so it only has a default export

I see, I had assumed it was ESM. I double-checked and made a separate repro, and realized the issue is a bit different than what I initially described. The particular problem I'm seeing is that ESLint's plugin normalization isn't working in the following scenario:

- Config B, in its own package, defines the import plugin
- Config A, in a different package, extends Config B
- Config A also defines the import plugin

The following repro illustrates this problem: [import-repro.zip](https://github.com/user-attachments/files/22568817/import-repro.zip)

I believe this is because the `plugin` object that the flat configs define is not referentially equal to the plugin's default export. My linked PR resolves this problem.

**Comment by ljharb** [comment]
Is that something that eslint requires? where is that documented?

**Comment by soren121** [comment]
Their [documentation for defining configurations](https://eslint.org/docs/latest/extend/plugins#configs-in-plugins) in the flat config system includes a code example that implies you are meant to use the same object that is exported, but they do not explicitly spell out that it _must_ be the same object. I don't find that too surprising, it's a fairly small edge case after all.

The specific line that throws this error in ESLint is here:
https://github.com/eslint/eslint/blob/fc9381f6ca57b824e82d118c14631c17bea79d7e/lib/config/flat-config-schema.js#L394

They do expect that the plugin objects are referentially equal if they are defined with the same name, otherwise it throws an error.

**Comment by sergio-milu** [comment]
this is still broken, as a workaround I did this


```
const compat = new FlatCompat({
  baseDirectory: import.meta.dirname,
});

const eslintImport = [
  ...compat.config({
    extends: ['plugin:import/recommended', 'plugin:import/typescript'],
  }),
];

export default defineConfig([
    ...eslintImport
])
```
